ten-simple-rules-dockerfiles.Rmd

```diff
@@ -562,70 +561,64 @@ RUN pip install \
-Such an installation from a source gives you much freedom, including the freedom to shoot yourself in the foot!
+Finally, you can use many package managers to install software from source obtained from external code management repositories, e.g., installing a tool from a specific GitHub branch or version tag.
+Be aware of the risk that such installations may later fail, especially when the external repositories are out of your control. However, these concerns can be mitigated by running the installation command with the full URL (including the specific version tag or commit hash) which is helpful in troubleshooting if problems arise.
 The version pinning capabilities of these file formats and package managers are described in their respective documentation.
 
-As a final note on software installation, you should be aware of the [`USER` instruction](https://docs.docker.com/engine/reference/builder/#user) in a `Dockerfile` and how your base image might change the user for particular instructions, which can impact by which user account some commands can be run within the container.
+As a final note on software installation, you should be aware of the [`USER` instruction](https://docs.docker.com/engine/reference/builder/#user) in a `Dockerfile` and how your base image might change the user for particular instructions, restricting which commands can be run within the container.
 It is common to use images with the default user `root`, which is required for installing system dependencies.
-We recommend making sure that the image works without specifying any users, and, if your image deviates from that, we suggest you document it precisely.
-One reason you will encounter base images running a non-root user (popular examples are the Jupyter and Rocker images) is to avoid permission problems when mounting files into the container, especially for "output" files (see&nbsp;\ruleref{rule:mount}).
+However you may encounter base images running as a non-root user (e.g., in the Jupyter and Rocker image stacks), in order to avoid permission problems when mounting files into the container, especially for "output" files (see&nbsp;\ruleref{rule:mount}).
+We recommend ensuring that the image works without specifying any users, and, if your image deviates from that, we suggest you document it precisely.
 
-# 7. Mount user scripts and data {-}
+# 7. Mount datasets at run time {-}
 \rulelabel{7}{rule:mount}
 
-The role of containers is to provide the computing environment, not to encapsulate datasets or custom scripts.
-It is better to include data files and custom scripts from the local machine in the container at runtime, and use the container image primarily for the software and dependencies.
+The role of containers is to provide the computing environment, not to encapsulate (potentially very large) datasets.
+It is better to insert data files from the local machine into the container at runtime, and use the image primarily for the software and dependencies.
```

Comment:
I would add the prefix "large data files" because for small data files, it's okay to put them in the container. For example, for my pokemon library I keep a "database" of metadata and images for the pokemon, and it allows it to run without a hitch. Another example is providing a small dummy dataset example, so a user running for the first time (without a data file) has one packaged with the container exactly as his/her's should look.

Reply:
Done!